java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -308,8 +308,23 @@ public Command build() {
                   "--max_idle_secs=10",
                   "--bazelrc=" + bazelRcPath));
 
-      command.addAll(args);
+      // This would split the args "run //target -- hello world" into
+      // "run //target" and "-- hello world" ("hello world" being passed to the executable
+      // to run).
+      int bazelArgsTerminator = args.indexOf("--");
+      List<String> bazelArgs;
+      List<String> nonBazelArgs;
+      if (bazelArgsTerminator == -1) {
+        bazelArgs = args;
+        nonBazelArgs = Collections.emptyList();
+      } else {
+        bazelArgs = args.subList(0, bazelArgsTerminator);
+        nonBazelArgs = args.subList(bazelArgsTerminator, args.size());
+      }
+
+      command.addAll(bazelArgs);
```

Comment:
what about
```java
int terminator = args.indexOf("--");
if (terminator) {
  command.AddAll(args);
  command.AddAll(repositoryCache.bazelOptions());
} else {
  command.AddAll(args.subList(0, terminator));
  command.AddAll(repositoryCache.bazelOptions());
  command.AddAll(args.subList(terminator, args.size()));
}
```

Reply:
I made room for options other than repositoryCache, but I like your solution better.